pkg/standalone-utils/contracts/LidoRelayer.sol

```diff
@@ -0,0 +1,210 @@
+            require(!funds.fromInternalBalance, "Cannot send from internal balance");
+
+            _pullStETHAndWrap(msg.sender, singleSwap.amount);
+            _approveToken(IERC20(address(_wstETH)), address(getVault()), singleSwap.amount);
+        } else if (singleSwap.assetOut == IAsset(address(_wstETH))) {
+            // If wstETH is an output then we want to receive it on the relayer
+            // so we can unwrap it before forwarding stETH to the user
+            funds.recipient = payable(address(this));
+            require(!funds.toInternalBalance, "Cannot send to internal balance");
+        } else {
+            revert("Does not require wstETH");
+        }
+
+        swapAmount = getVault().swap{ value: msg.value }(singleSwap, funds, limit, deadline);
+
+        if (singleSwap.assetOut == IAsset(address(_wstETH))) {
+            // Unwrap any received wstETH for the user automatically
+            _unwrapAndPushStETH(recipient, swapAmount);
+        } else if (singleSwap.kind == IVault.SwapKind.GIVEN_OUT) {
+            // GIVEN_OUT swaps with wstETH input may leave some dust on the relayer
+            // This should be forwarded on to the user
+            _unwrapAndPushStETH(recipient, IERC20(address(_wstETH)).balanceOf(address(this)));
+        }
+
+        _sweepETH();
+    }
+
+    function batchSwap(
+        IVault.SwapKind kind,
+        IVault.BatchSwapStep[] calldata swaps,
+        IAsset[] calldata assets,
+        IVault.FundManagement memory funds,
+        int256[] calldata limits,
+        uint256 deadline
+    ) external payable nonReentrant returns (int256[] memory swapAmounts) {
+        require(funds.sender == msg.sender, "Invalid sender");
+        // Cache recipient as we sometimes overwrite this
+        address recipient = funds.recipient;
+
+        // Find the index of wstETH in the assets array
+        uint256 wstETHIndex;
+        for (uint256 i; i < assets.length; i++) {
+            if (assets[i] == IAsset(address(_wstETH))) {
+                wstETHIndex = i;
+                break;
+            }
+            require(i < assets.length, "Does not require wstETH");
+        }
+
+        int256 wstETHLimit = limits[wstETHIndex];
+        if (wstETHLimit > 0) {
+            // If wstETH is an input then we want to send it from the relayer
+            // as we wrap it there.
+            funds.sender = address(this);
+            require(!funds.fromInternalBalance, "Cannot send from internal balance");
+
+            _pullStETHAndWrap(msg.sender, uint256(wstETHLimit));
+            _approveToken(IERC20(address(_wstETH)), address(getVault()), uint256(wstETHLimit));
+        } else {
+            // If wstETH is an output then we want to receive it on the relayer
+            // so we can unwrap it before forwarding stETH to the user
+            funds.recipient = payable(address(this));
+            require(!funds.toInternalBalance, "Cannot send to internal balance");
```

Comment:
Similarly, this assumes the only output token is wstETH. We should be explicit about this in the documentation.

Reply:
I've added a couple of lines on the risks of multiple inputs/outputs.